### PR TITLE
WIP add WHERE clause generator for skinny table

### PR DESCRIPTION
**Pull request by wavemechanics**
### What

Added WHERE clause generator in preparation for querying skinny table.

### CODE REVIEW

File: pkg/demo/demo.go [hunk 1]
```diff
@@ -85,12 +85,28 @@ func New(pgpwd string) (*Demo, error) {
 	return app, nil
 }
 
+func (app *Demo) Query(ctx context.Context, dataset string, rows, cols []string) (string, error) {
+	if dataset == "skinny" {
```
Comment (VivianAllen) [inline pkg/demo/demo.go:89]: should this name be hardcoded?

File: pkg/demo/demo.go [hunk 2]
```diff
@@ -85,12 +85,28 @@ func New(pgpwd string) (*Demo, error) {
...
+	return clause, nil
+}
+
 // query runs a SQL query against the db and returns the resulting CSV as a string.
 // dataset is the name of the table to query.
 // rowspec is not used at present.
 // colspec is a list of columns to include in the result. Empty means all columns.
 //
-func (app *Demo) Query(ctx context.Context, dataset string, rowspec, colspec []string) (string, error) {
+func (app *Demo) tableQuery(ctx context.Context, dataset string, rowspec, colspec []string) (string, error) {
```
Comment (VivianAllen) [inline pkg/demo/demo.go:109]: Realise its WIP, but the skinnyQuery / tableQuery naming seems a little off to me. Can't immediately think of better options though.

File: pkg/where/skinny.go [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+//
+// Because of the size of the geo_metrics table, we require at least one geography_code
+// and at least one category_code to be specified.
+//
+//
+
+// SkinnyWhere generates the WHERE clause for queries against the geo_metrics table.
+// rows and cols are the multivalue rows= and cols= arguments.
+//
+func SkinnyWhere(rows, cols []string) (string, error) {
```
Comment (VivianAllen) [inline pkg/where/skinny.go:33]: is 'vertically mixed table' better than 'skinny table'?

File: pkg/demo/demo.go (hunk 1 above)
Comment (wavemechanics) [reply pkg/demo/demo.go:89]: We don't necessarily have to do it this way. There only has to be a mechanism to switch between the two types of queries, and this was easy.
(The 'skinny' query does the pivot from a known table, and the other queries (so far) are normal row and column selects.)

File: pkg/demo/demo.go (hunk 2 above)
Comment (wavemechanics) [reply pkg/demo/demo.go:109]: yep, open to suggestions

File: pkg/where/skinny.go (hunk 3 above)
Comment (wavemechanics) [reply pkg/where/skinny.go:33]: I think we can rationalise the naming as we add different queries. Definitely needs work.

File: pkg/where/skinny.go [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+//
+func WherePart(col string, args []string) (string, error) {
+	var conditions []string
+
+	set, err := GetValues(args)
+	if err != nil {
+		return "", err
+	}
+
+	if len(set.Singles) > 0 {
```
Comment (VivianAllen) [inline pkg/where/skinny.go:63]: any point in an early return here if this condition isn't met?

File: pkg/demo/demo.go (hunk 1 above)
Comment (VivianAllen) [reply pkg/demo/demo.go:89]: maybe some kind of `pivot` parameter?

**Review by VivianAllen — APPROVED** [review]
as discussed in comments I think theres some naming stuff that still needs to get nailed down, but this is looking really nice.

**Pull request merged**